src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,117 @@ class JoinClauseSegment(BaseSegment):
+        # Nested Selects
+        Bracketed(Ref("SelectableGrammar")),
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                Ref("SingleIdentifierGrammar", optional=True),
+                # Column alias in VALUES clause
+                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
+            ),
+            Ref("QuotedLiteralSegment"),
+        ),
+    )
+
+
+@spark3_dialect.segment()
+class DelimitedValues(BaseSegment):
+    """A ``VALUES`` clause can be a sequence either of scalar values or tuple values.
+
+    We make no attempt to ensure that all records have the same number of columns besides the distinction between
+    all scalar or all tuple, so for instance ``VALUES (1,2), (3,4,5)`` will parse but is not legal SQL.
+
+
+    """
+
+    type = "delimited_values"
+    match_grammar = OneOf(Delimited(Ref("ScalarValue")), Delimited(Ref("TupleValue")))
+
+
+@spark3_dialect.segment()
+class ScalarValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has a single column.
+
+    Ex: ``VALUES 1,2,3``
+    """
+
+    type = "scalar_value"
+    match_grammar = OneOf(
+        Ref("LiteralGrammar"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+    )
+
+
+@spark3_dialect.segment()
+class TupleValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has a multiple columns.
+
+    Ex: ``VALUES (1,2), (3,4)``
+    """
+
+    type = "tuple_value"
+    match_grammar = Bracketed(
+        Delimited(
+            Ref("ScalarValue"),
+        )
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class ValuesClauseSegment(BaseSegment):
+    """A ``VALUES`` clause, as typically used with ``INSERT`` or ``SELECT``.
+
+    The Spark SQL reference does not mention ``VALUES``` clauses except in the context of ``INSERT`` statements.
+    However, they appear to behave the same as in `postgres <https://www.postgresql.org/docs/13/sql-values.html>`.
+
+    In short, they can appear anywhere a ``SELECT`` can, and also as bare ``VALUES`` statements. Here are some examples:
+    .. code-block:: sql
+
+        VALUES 1,2 LIMIT 1;
+        SELECT * FROM VALUES (1,2) as t (a,b);
+        SELECT * FROM (VALUES (1,2) as t (a,b));
+        WITH a AS (VALUES 1,2) SELECT * FROM a;
+
+    """
+
+    type = "values_clause"
+    match_grammar = StartsWith(Ref.keyword("VALUES"))
+
+    parse_grammar = Sequence(
```

Comment:
I have to confess I do not understand the difference between `match_grammar` and `parse_grammar`, so this is just parroting a similar segment I saw elsewhere.

Reply:
`match_grammar` is like a "hint" to the parser to help it distinguish when a segment starts or stops. `parse_grammar` is a full grammar, i.e. describes the entire structure. Often, the `match_grammar` is not needed. Generally, you consider adding one if the parser is failing to parse some of the test SQL. If things are working okay like this, I would go ahead and keep it.

Reply:
interesting, i had noticed `parse_grammar` not being defined but not seen a place where there was no `match_grammar`

but you're saying i could generally omit `match_grammar` and only define it if i run into trouble?

Reply:
Maybe I had it backwards. I remembered you could omit one of them. So I think if you only provide `match_grammar`, it uses the same for both.

Reply:
Exactly. You always need a `match_grammar` and it is used to parse the statement if no dedicated `parse_grammar` is provided. It's discussed here: https://docs.sqlfluff.com/en/stable/architecture.html#stage-3-the-parser

As @barrywhart says, `match_grammar` is useful if you can have a simple shortcut to identify the start and end of this grammar. It can then be parsed. Rather than trying to parse with no ending in sight, where it may overshot into the next statement for certain combinations.

Here you are only specifying the start (it starts with `"VALUES"`) so there is no benefit to splitting them out. There was some discussion as to whether that was quicker (i.e. did the parse need to load the whole, potentially complicated statement, then check first word?) but tests showed it's exactly the same.

So merge `match_grammar` and `parse_grammar` here into the one `match_grammar` definition with the full definition as no benefit in it being split.

Reply:
ahh thanks that clears things up, i'll rtfm